akvo/rest/serializers/indicator_period_data.py

```diff
@@ -73,7 +73,14 @@ def update(self, instance, validated_data):
             )
             serializer.update(disaggregation_instance, serializer.validated_data)
 
-        return instance
+        return instance._meta.model.objects.select_related(
```

Comment:
Good catch!

Reply:
by the tests. But, nevertheless, good. 👍